center/src/main/java/com/microsoft/hydralab/center/service/AgentManageService.java

```diff
@@ -128,4 +130,38 @@ public void updateAgentTeam(String teamId, String teamName){
 
          agentUserRepository.saveAll(agents);
     }
+
+    public Boolean downloadAgentConfigFile(SysUser requestor, String agentId) {
+        AgentUser agentUser = getAgent(agentId);
+        if (agentUser != null) {
+            try {
+                File agentConfigFile = File.createTempFile(
+                        "application",
+                        ".yml",
+                        new File(".\\"));
+
+                FileWriter fileWriter = new FileWriter(agentConfigFile.getAbsolutePath());
+                fileWriter.write("app:\n" +
+                        "  # register to Hydra Lab Center\n" +
+                        "  registry:\n" +
+                        "    # The server hostname:port of Hydra Lab Center. If nginx enabled, switch to port of nginx\n" +
+                        "    server: 'localhost:9886'\n" +
+                        "    # The Agent info registered in Hydra Lab Center, for instance if it's running on localhost, the URL would be: http://localhost:9886/portal/index.html#/auth\n" +
+                        "    name: " + agentUser.getName() + "\n" +
+                        "    id: " + agentUser.getId() + "\n" +
+                        "    secret: " + agentUser.getSecret() + "\n" +
+                        "    # Agent Type {1 : 1*WINDOWS + n*ANDROIDS , 2 : 1*WINDOWS+1*ANDROID , 3 : iOS}\n" +
+                        "    agent-type: 1");
+                fileWriter.flush();
```

Comment:
The fileWriter should be close after flush. Otherwise, the file would be using status.